### PR TITLE
refactor[ux]: add `venom` as `experimental-codegen` alias

### PR DESCRIPTION
**Pull request by sandbubbles**
### What I did
Make a "venom" alias to "experimental_codegen" flag.
See https://github.com/vyperlang/vyper/issues/4334.

### How I did it
Add the alias to cli, pre-parser, and json parser.
### How to verify it
All exisiting tests work with old flas as well as new tests with `venom`.

### Commit message

```
Make "venom" an alternative alias to the "experimental_codegen" flag.
Add the alias to cli, pre-parser, and json parser.
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/cli/vyper_json.py [hunk 1]
```diff
@@ -254,6 +254,8 @@ def get_settings(input_dict: dict) -> Settings:
 
     optimize = input_dict["settings"].get("optimize")
     experimental_codegen = input_dict["settings"].get("experimentalCodegen")
+    if experimental_codegen is None:
+        experimental_codegen = input_dict["settings"].get("venom")
```
Comment (charles-cooper) [inline vyper/cli/vyper_json.py:258]: Maybe throw an exception if both are set

**Review by charles-cooper — COMMENTED** [review]
Thanks! Can we add tests for conflicts - when both are set

File: tests/unit/ast/test_pre_parser.py [hunk 2]
```diff
@@ -191,6 +191,25 @@ def test_parse_pragmas(code, pre_parse_settings, compiler_data_settings, mock_ve
     assert compiler_data.settings == compiler_data_settings
 
 
+pragma_venom = [
+    """
+    #pragma venom
```
Comment (charles-cooper) [inline tests/unit/ast/test_pre_parser.py:196]: can we add a test for when both are set?

Comment (sandbubbles) [reply tests/unit/ast/test_pre_parser.py:196]: forgot to commit the file, sorry! should be there now

File: tests/unit/ast/test_pre_parser.py [hunk 3]
```diff
@@ -218,6 +237,15 @@ def test_parse_pragmas(code, pre_parse_settings, compiler_data_settings, mock_ve
 # pragma evm-version cancun
 # pragma evm-version shanghai
     """,
+    # duplicit setting of venom
```
Comment (charles-cooper) [inline tests/unit/ast/test_pre_parser.py:240]: typo: duplicate

File: tests/unit/ast/test_pre_parser.py (hunk 2 above)
Comment (charles-cooper) [reply tests/unit/ast/test_pre_parser.py:196]: thanks!

File: tests/unit/cli/vyper_json/test_compile_json.py [hunk 4]
```diff
@@ -319,3 +320,37 @@ def test_compile_json_with_abi_top(make_input_bundle):
...
+            "evmVersion": "cancun",
+            "optimize": "gas",
+            "experimentalCodegen": False,
+            "venom": False,
+            "search_paths": [],
+            "outputSelection": {"*": ["ast"]},
+        },
+    }
+    with pytest.raises(JSONError):
+        get_settings(code)
```
Comment (charles-cooper) [inline tests/unit/cli/vyper_json/test_compile_json.py:356]: maybe check the error message
```suggestion
    with pytest.raises(JSONError):
        get_settings(code)
    assert "both experimentalCodegen and venom cannot be set" in str(e.value)
```

Comment (charles-cooper) [reply tests/unit/cli/vyper_json/test_compile_json.py]: https://github.com/vyperlang/vyper/pull/4337/commits/d6bba9917f4e168c0005427a110e6d615cf732e4

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**